src/specs/VideoNativeComponent.ts

```diff
@@ -0,0 +1,563 @@
+  cropEnd?: Float;
+  title?: string;
+  subtitle?: string;
+  description?: string;
+  customImageUri?: string;
+}>;
+
+export type DRMType = WithDefault<
+  'widevine' | 'playready' | 'clearkey' | 'fairplay',
+  'widevine'
+>;
+
+type DebugConfig = Readonly<{
+  enable?: boolean;
+  thread?: boolean;
+}>;
+
+type Drm = Readonly<{
+  type?: DRMType;
+  licenseServer?: string;
+  headers?: Headers;
+  contentId?: string; // ios
+  certificateUrl?: string; // ios
+  base64Certificate?: boolean; // ios default: false
+  useExternalGetLicense?: boolean; // ios
+}>;
+
+type TextTracks = ReadonlyArray<
+  Readonly<{
+    title: string;
+    language: string;
+    type: string;
+    uri: string;
+  }>
+>;
+
+type SelectedTrack = WithDefault<
+  'system' | 'disabled' | 'title' | 'language' | 'index',
+  'system'
+>;
+
+export type SelectedTrackType = Readonly<{
+  type?: SelectedTrack;
+  value?: string;
+}>;
+
+export type Seek = Readonly<{
+  time: Float;
+  tolerance?: Float;
+}>;
+
+type BufferConfig = Readonly<{
+  minBufferMs?: Float;
+  maxBufferMs?: Float;
+  bufferForPlaybackMs?: Float;
+  bufferForPlaybackAfterRebufferMs?: Float;
+  maxHeapAllocationPercent?: Float;
+  minBackBufferMemoryReservePercent?: Float;
+  minBufferMemoryReservePercent?: Float;
+}>;
+
+type SelectedVideoTrack = WithDefault<
+  'auto' | 'disabled' | 'resolution' | 'index',
+  'auto'
+>;
+
+export type SelectedVideoTrackType = Readonly<{
+  type?: SelectedVideoTrack;
+  value?: Int32;
+}>;
+
+type SubtitleStyle = Readonly<{
+  fontSize?: Float;
+  paddingTop?: WithDefault<Float, 0>;
+  paddingBottom?: WithDefault<Float, 0>;
+  paddingLeft?: WithDefault<Float, 0>;
+  paddingRight?: WithDefault<Float, 0>;
+}>;
+
+export type OnLoadData = Readonly<{
+  currentTime: Float;
+  duration: Float;
+  naturalSize: Readonly<{
+    width: Float;
+    height: Float;
+    orientation: WithDefault<'landscape' | 'portrait', 'landscape'>;
+  }>;
+  audioTracks: Readonly<{}>;
+  textTracks: Readonly<{}>;
+}>;
+
+export type OnLoadStartData = Readonly<{
+  isNetwork: boolean;
+  type: string;
+  uri: string;
+}>;
+
+export type OnVideoAspectRatioData = Readonly<{
+  width: Float;
+  height: Float;
+}>;
+
+export type OnBufferData = Readonly<{isBuffering: boolean}>;
+
+export type OnProgressData = Readonly<{
+  currentTime: Float;
+  playableDuration: Float;
+  seekableDuration: Float;
+}>;
+
+export type OnBandwidthUpdateData = Readonly<{
+  bitrate: Int32;
+  width?: Float;
+  height?: Float;
+  trackId?: Int32;
+}>;
+
+export type OnSeekData = Readonly<{
+  currentTime: Float;
+  seekTime: Float;
+}>;
+
+export type OnPlaybackStateChangedData = Readonly<{
+  isPlaying: boolean;
+}>;
+
+export type OnTimedMetadataData = Readonly<{
+  metadata: Readonly<{}>;
+}>;
+// TODO fix type after RN 0.73
+// export type OnTimedMetadataData = Readonly<{
+//   metadata: ReadonlyArray<
+//     Readonly<{
+//       value?: string;
+//       identifier: string;
+//     }>
+//   >;
+// }>;
```

Comment:
What do you mean by "fix type after RN 0.73" ? 0.73 is already out 🤔

Reply:
oh that means, we have to bump up "react-native" version on top-level's package.json to above 0.73.
Right now, it's below 0.73. (in devDependencies)

So if we want to support array parameters on event callback functions via codegen, I think we can only fully support fabric after RN 0.73 version.

Reply:
But right now, this PR is only for support codegen.
Not implemented native codes yet.
So I just commented out and didn't consider react-native 0.73 on this PR.

Reply:
But, I think it's my bad. It'll affect JS layer's event prop types too.
I will make a fix commit soon.

Reply:
@KrzysztofMoch 
I think these two commits will fix this issue.
https://github.com/react-native-video/react-native-video/pull/3487/commits/0c4bd9ce749d1e6c5f501e2ce4901836bc326314
https://github.com/react-native-video/react-native-video/pull/3487/commits/c9f6a9254592e66c3b9382aa2bebf35dc1ed7c10